clients/drcachesim/tools/schedule_stats.h

```diff
@@ -112,6 +157,7 @@ class schedule_stats_t : public analysis_tool_t {
         uint64_t cpu_microseconds = 0;
         uint64_t wait_microseconds = 0;
         std::unordered_set<memref_tid_t> threads;
+        std::unique_ptr<histogram_t> instrs_per_switch;
```

Comment:
If a user wants a more sophisticated histogram_t, I guess they'd just over-write this field with their other implementation? Cleanup should just work since this is a unique_ptr.

I suppose a nicer way to allow substituting a more sophisticated histogram_t would be to have a virtual create_histogram function that can be overridden by classes derived from counters_t. But that does create more complexity of creating a sub-class for counters_t.

Fine either way.

Reply:
I was thinking there must be a better way but with the counters_t extra layer it seems like any approach has a little complexity.  Going to go with what's there now.